lib/query.js

```diff
@@ -4889,6 +4890,15 @@ Query.prototype.exec = function exec(op, callback) {
   // Ensure that `exec()` is the first thing that shows up in
   // the stack when cast errors happen.
   const castError = new CastError();
+  const collection = this.mongooseCollection;
+
+  if (
```

Comment:
Is this necessary? The `strictQuery` warning will print whenever you instantiate a connection, which (1) should always happen, and (2) is better behavior IMO - prints when process starts for most apps, as opposed to at runtime.

Reply:
I agree that it is better to display the warning only on the connection.
It looks a bit redundant here unless there are cases to handle that I'm unaware of.
If it's ok for you I'll submit a commit with the `printStrictQueryWarning` only in `index.js`'s `connect` method.